File: Client/hms/src/app/services/inventory.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { FilterObject } from '../models/FilterObject';
import { ApiResponse } from '../models/ApiResponse';
import { Inventory } from '../models/Inventory';

@Injectable({
  providedIn: 'root'
})
export class InventoryService {

  constructor(private httpClient:HttpClient) { }

  private apiBaseUrl : string = "http://localhost:8888";


  addOrUpdate(inventory:Inventory){
    return this.httpClient.post<ApiResponse<Inventory>>(this.apiBaseUrl+"/inventory/",inventory);
  }
  getInventory(id:number){
    return this.httpClient.get<ApiResponse<Inventory>>(this.apiBaseUrl+"/inventory/byId/"+id);
  }
  searchInventory(filterObject:FilterObject){
    return this.httpClient.post<ApiResponse<Inventory>>(this.apiBaseUrl+"/inventory/search",filterObject);
  }
 
}

File: Server/hms/src/main/java/com/hms/inventory/service/InventoryService.java
package com.hms.inventory.service;

import com.hms.common.dto.CommonListTO;
import com.hms.common.dto.FilterObject;
import com.hms.inventory.dto.InventoryTO;

public interface InventoryService {
    InventoryTO saveOrUpdate(InventoryTO inventory);
    InventoryTO byId(Integer id);
    CommonListTO<InventoryTO> search(FilterObject filterObject);

}
